### PR TITLE
fix cloudcore cluster role

### PR DESCRIPTION
**Pull request by wackxu**
Signed-off-by: wackxu <xushiwei5@huawei.com>

**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: manifests/charts/cloudcore/templates/rbac_cloudcore.yaml [hunk 1]
```diff
@@ -14,7 +14,7 @@ rules:
   resources: ["namespaces"]
   verbs: ["get", "create", "list", "watch"]
 - apiGroups: [""]
-  resources: ["nodes", "pods/status"]
+  resources: ["nodes", "nodes/status", "pods/status"]
```
Comment (fisherxu) [inline manifests/charts/cloudcore/templates/rbac_cloudcore.yaml:17]: The CI haven't reported the error? cc @gy95

Comment (gy95) [reply manifests/charts/cloudcore/templates/rbac_cloudcore.yaml:17]: keadm e2e contains `Create a pod and check the pod is coming up correctly`
https://github.com/kubeedge/kubeedge/blob/master/tests/e2e/keadm/keadm_test.go
maybe not cover
how to trigger this? where do we need patch methods for nodes/status?

Comment (wackxu) [reply manifests/charts/cloudcore/templates/rbac_cloudcore.yaml:17]: https://github.com/kubeedge/kubeedge/blob/340a251b44d5d8e28061758e86345f81190882f3/cloud/pkg/edgecontroller/controller/upstream.go#L968 in new edged, which update the node status

Comment (wackxu) [reply manifests/charts/cloudcore/templates/rbac_cloudcore.yaml:17]:
```
Sep 26 14:52:43 wackxu-dev edgecore[3611914]: E0926 14:52:43.030707 3611914 kubelet_node_status.go:400] "Error updating node status, will retry" err="failed to patch status \"{\\\"status\\\":{\\\"$setElementOrder/conditions\\\":[{\\\"type\\\":\\\"MemoryPressure\\\"},{\\\"type\\\":\\\"DiskPressure\\\"},{\\\"type\\\":\\\"PIDPressure\\\"},{\\\"type\\\":\\\"Ready\\\"}],\\\"conditions\\\":[{\\\"lastHeartbeatTime\\\":\\\"2022-09-26T06:52:43Z\\\",\\\"type\\\":\\\"MemoryPressure\\\"},{\\\"lastHeartbeatTime\\\":\\\"2022-09-26T06:52:43Z\\\",\\\"type\\\":\\\"DiskPressure\\\"},{\\\"lastHeartbeatTime\\\":\\\"2022-09-26T06:52:43Z\\\",\\\"type\\\":\\\"PIDPressure\\\"},{\\\"lastHeartbeatTime\\\":\\\"2022-09-26T06:52:43Z\\\",\\\"type\\\":\\\"Ready\\\"}]}}\" for node \"e2e-test\": nodes \"e2e-test\" is forbidden: User \"system:serviceaccount:kubeedge:cloudcore\" cannot patch resource \"nodes/status\" in API group \"\" at the cluster scope"

Comment (fisherxu) [reply manifests/charts/cloudcore/templates/rbac_cloudcore.yaml:17]: > keadm e2e contains `Create a pod and check the pod is coming up correctly` https://github.com/kubeedge/kubeedge/blob/master/tests/e2e/keadm/keadm_test.go maybe not cover how to trigger this? where do we need patch methods for nodes/status?

IIRC we already have a test in keadm_e2e which builds the container from source code? If the rule `nodes/status` missed, the node status can't be ready?

Comment (wackxu) [reply manifests/charts/cloudcore/templates/rbac_cloudcore.yaml:17]: in my environment that use keadm install , the node status is always not ready

```
root@open-source-dev:~/test-edge-pod# kubectl get node
NAME        STATUS     ROLES        AGE     VERSION
127.0.0.1   Ready      <none>       3h48m   v1.22.6-dirty
e2e-test    NotReady   agent,edge   14m     v1.22.6-kubeedge-v1.11.0-beta.0.211+347278db67ea3d-dirty

```

Comment (gy95) [reply manifests/charts/cloudcore/templates/rbac_cloudcore.yaml:17]: https://github.com/kubeedge/kubeedge/blob/340a251b44d5d8e28061758e86345f81190882f3/tests/e2e/scripts/keadm_e2e.sh#L75-L79
maybe this check is not correct, just like https://github.com/kubeedge/kubeedge/pull/4231

Comment (fisherxu) [reply manifests/charts/cloudcore/templates/rbac_cloudcore.yaml:17]: > maybe this check is not correct, just like https://github.com/kubeedge/kubeedge/pull/4231

If not correct, the test case should not passed :)

Comment (gy95) [reply manifests/charts/cloudcore/templates/rbac_cloudcore.yaml:17]: yes, by test in https://github.com/kubeedge/kubeedge/pull/4235, it indeed is

Comment (gy95) [reply manifests/charts/cloudcore/templates/rbac_cloudcore.yaml:17]: can fix like https://github.com/kubeedge/kubeedge/pull/4235,
`grep -q Ready` -> `grep -q -w Ready`

Comment (fisherxu) [reply manifests/charts/cloudcore/templates/rbac_cloudcore.yaml:17]: Besides the node `notready` status, the app seems have no node-selector? Or how the app can run now.

**Review by fisherxu — COMMENTED** [review]
/lgtm
/approve

**Comment by fisherxu** [comment]
Let's cherry-pick to release-1.12 :)

**Pull request merged**